tests/CMakeLists.txt

```diff
@@ -151,51 +334,84 @@ target_compile_definitions(ssh_client_test PRIVATE
 target_compile_definitions(utils_test PRIVATE
   -DEVP_PBE_scrypt=ut_EVP_PBE_scrypt)
 
-target_link_libraries(multipass_tests
-  cert
-  client
-  daemon
-  delayed_shutdown
-  gmock_main
-  gtest_main
-  ip_address
-  iso
-  network
-  petname
+# Base libraries needed by all tests
+set(BASE_TEST_LIBRARIES
+  client # For shared functionality
+  gmock_main # For mocking
+  gtest_main # For testing framework
+  premock # For mocking
+  scope_guard # For cleanup
+  utils_test # For utility functions
+)
+
+# Minimal networking libraries
+set(MEMORY_TEST_LIBRARIES
```

Comment:
To avoid clutter, we should move non-common library lists to their respective target_link_libraries instead.